src/core/annotation.js

```diff
@@ -746,6 +739,37 @@ var LinkAnnotation = (function LinkAnnotationClosure() {
   return LinkAnnotation;
 })();
 
+var PopupAnnotation = (function PopupAnnotationClosure() {
+  function PopupAnnotation(parameters) {
+    Annotation.call(this, parameters);
+
+    this.data.annotationType = AnnotationType.POPUP;
+
+    var dict = parameters.dict;
+    if (!dict.has('Parent')) {
+      warn('Popup annotations without a parent annotation are not supported.');
+      return;
+    }
+
+    var parentItem = dict.get('Parent');
+    if (!parentItem) {
+      warn('Popup annotation has an invalid parent annotation.');
+      return;
+    }
+    this.data.parentId = dict.map.Parent.num;
```

Comment:
Please note that you should *not* access members of a `Dict` by directly using `map` (it should be considered private), use e.g. [`Dict_get`](https://github.com/mozilla/pdf.js/blob/master/src/core/primitives.js#L103) instead.